apps/quick-learn-backend/src/routes/lesson/lesson.service.ts

```diff
@@ -204,4 +208,33 @@ export class LessonService extends PaginationService<LessonEntity> {
       },
     );
   }
+
+  /**
+   * Permanently deletes a lesson
+   * @param id - The id of the lesson to delete
+   * @throws BadRequestException if the lesson doesn't exist
+   */
+  async deleteLesson(id: number): Promise<void> {
```

Comment:
A transaction is not needed here. It is needed only when performing multiple operations, and two or more are related to each other.